source/slang/glsl.meta.slang

```diff
@@ -7263,3 +7263,704 @@ __glsl_version(430) [require(glsl)]
+{
+    __target_switch
+    {
+    case hlsl:
+    {
+        return abs(ddx_fine(p)) + abs(ddy_fine(p));
+    }
+    case glsl: __intrinsic_asm "fwidthFine($0)";
+    case spirv:
+    {
+        return spirv_asm
+        {
+            OpCapability DerivativeControl;
+            OpFwidthFine $$vector<float, N> result $p;
+        };
+    }
+    }
+}
+
+[require(hlsl, sm_4_0)]
+__glsl_version(430) [require(glsl)]
+__spirv_version(1.0) [require(spirv)]
+[__NoSideEffect]
+[ForceInline]
+public float fwidthCoarse(float p)
+{
+    __target_switch
+    {
+    case hlsl:
+    {
+        return abs(ddx_coarse(p)) + abs(ddy_coarse(p));
+    }
+    case glsl: __intrinsic_asm "fwidthCoarse($0)";
+    case spirv:
+    {
+        return spirv_asm
+        {
+            OpCapability DerivativeControl;
+            OpFwidthCoarse $$float result $p;
+        };
+    }
+    }
+}
+[require(hlsl, sm_4_0)]
+__glsl_version(430) [require(glsl)]
+__spirv_version(1.0) [require(spirv)]
+__generic<let N : int>
+[__NoSideEffect]
+[ForceInline]
+public vector<float, N> fwidthCoarse(vector<float, N> p)
+{
+    __target_switch
+    {
+    case hlsl:
+    {
+        return abs(ddx_coarse(p)) + abs(ddy_coarse(p));
+    }
+    case glsl: __intrinsic_asm "fwidthCoarse($0)";
+    case spirv:
+    {
+        return spirv_asm
+        {
+            OpCapability DerivativeControl;
+            OpFwidthCoarse $$vector<float, N> result $p;
+        };
+    }
+    }
+}
+
+[require(hlsl, sm_5_0)]
+__glsl_version(430) [require(glsl)]
+__spirv_version(1.0) [require(spirv)]
+[__NoSideEffect]
+[GLSLRequireShaderInputParameter(0)]
+public float interpolateAtCentroid(__ref float interpolant)
+{
+    __target_switch
+    {
+    case hlsl:
+    {
+        return EvaluateAttributeAtCentroid(interpolant);
+    }
+    case glsl: __intrinsic_asm "interpolateAtCentroid($0)";
+    case spirv:
+    {
+        return spirv_asm
+        {
+            OpCapability InterpolationFunction;
+            OpExtInst $$float result glsl450 InterpolateAtCentroid &interpolant
+        };
+    }
+    }
+}
+[require(hlsl, sm_5_0)]
+__glsl_version(430) [require(glsl)]
+__spirv_version(1.0) [require(spirv)]
+__generic<let N : int>
+[__NoSideEffect]
+[GLSLRequireShaderInputParameter(0)]
+public vector<float, N> interpolateAtCentroid(__ref vector<float, N> interpolant)
+{
+    __target_switch
+    {
+    case hlsl:
+    {
+        return EvaluateAttributeAtCentroid(interpolant);
+    }
+    case glsl: __intrinsic_asm "interpolateAtCentroid($0)";
+    case spirv:
+    {
+        return spirv_asm
+        {
+            OpCapability InterpolationFunction;
+            OpExtInst $$vector<float, N> result glsl450 InterpolateAtCentroid &interpolant
+        };
+    }
+    }
+}
+
+[require(hlsl, sm_5_0)]
+__glsl_version(430) [require(glsl)]
+__spirv_version(1.0) [require(spirv)]
+[__NoSideEffect]
+[GLSLRequireShaderInputParameter(0)]
+public float interpolateAtSample(__ref float interpolant, int sample)
```

Comment:
This is defined for fragment, geometry, and other graphics stages, but not vertex, compute, raygen, miss etc. stages, so we need to reflect this in capability requirements.

Reply:
I did not add any capabilities for stages, I did not know these were in Slang, I will do this right-now